Sources/KeyboardSupport/KeyboardRespondable.swift

```diff
@@ -184,35 +195,50 @@ public extension KeyboardScrollable where Self: UIViewController {
         DispatchQueue.main.async {
             guard let textRange = textView.selectedTextRange, let selectionRect = textView.selectionRects(for: textRange).first else { return }
             // Set an arbitray width to the target CGRect in case the width is zero. Otherwise, scrollRectToVisible has no effect.
-            let contentRect = textView.convert(selectionRect.rect, to: scrollView).modifying(width: 30)
-            self.scrollToContentRectIfNecessary(contentRect: contentRect, keyboardInfo: keyboardInfo)
+            self.scrollToRectIfNecessary(rect: selectionRect.rect.modifying(width: 30), of: textView, keyboardInfo: keyboardInfo)
         }
     }
     
-    private func scrollToContentRectIfNecessary(contentRect: CGRect, keyboardInfo: KeyboardInfo) {
-        let keyboardMinY = view.bounds.height - keyboardInfo.finalFrame.height
-        guard let convertedTargetFrame = keyboardScrollableScrollView?.convert(contentRect, to: view), convertedTargetFrame.maxY > keyboardMinY else { return }
+    private func scrollToRectIfNecessary(rect: CGRect, of coordinateSpaceView: UIView, keyboardInfo: KeyboardInfo, preferredPaddingAroundInput: UIEdgeInsets? = nil) {
+        guard let scrollView = keyboardScrollableScrollView else { return }
         
-        UIView.animate(withDuration: keyboardInfo.animationDuration, delay: 0, options: [UIView.AnimationOptions(rawValue: keyboardInfo.animationCurve)], animations: {
-            self.keyboardScrollableScrollView?.scrollRectToVisible(convertedTargetFrame, animated: false)
+        // Determine padding
+        var paddingAroundInput: UIEdgeInsets = preferredPaddingAroundInput ?? .zero
+        paddingAroundInput = UIEdgeInsets(top: max(paddingAroundInput.top, minimumPaddingAroundInput.top),
+                                          left: max(paddingAroundInput.left, minimumPaddingAroundInput.left),
+                                          bottom: max(paddingAroundInput.bottom, minimumPaddingAroundInput.bottom),
+                                          right: max(paddingAroundInput.right, minimumPaddingAroundInput.right))
+        
+        //Inflate the frame being scrolled into view by the padding
+        let paddedFrameOfFirstResponder = rect.modifying(minY: rect.minY - paddingAroundInput.top)
+            .modifying(minX: rect.minX - paddingAroundInput.left)
+            .modifying(height: rect.height + paddingAroundInput.top + paddingAroundInput.bottom)
+            .modifying(width: rect.width + paddingAroundInput.left + paddingAroundInput.right)
+        
+        // Convert the padded rect to the scrollview coordinate space and scroll it into view
+        let paddedFrameOfFirstResponderInScrollView = coordinateSpaceView.convert(paddedFrameOfFirstResponder, to: scrollView)
+        UIView.animate(withDuration: keyboardInfo.animationDuration, delay: 0, options: .curveEaseInOut, animations: {
```

Comment:
Instead of hard-coding to the `.curveEaseInOut`, let's use the `animationCurve` property on `KeyboardInfo`.

Reply:
https://github.com/BottleRocketStudios/iOS-KeyboardSupport/pull/36/commits/c15289d6d82feb5652c6facd8761d0d8fc940f69